lib/Doctrine/DBAL/Connection.php

```diff
@@ -594,11 +588,13 @@ public function isConnected()
     /**
      * Checks whether a transaction is currently active.
      *
+     * @deprecated use getTransactionManager()->isTransactionActive()
```

Comment:
Instead of depending on one interface, the caller will now depend on two (Law of Demeter violation).

Reply:
I actually noticed this when trying to replace calls to these methods throughout the code, it's also very verbose to write and is definitely a code smell. As I said above, I'm not really at ease with moving the transaction manager code to the connection either as it breaks separation of concerns, but maybe it's the lesser of two evils...